Source/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry_CPPException.mm

```diff
@@ -92,6 +92,47 @@ void __cxa_throw(void *thrown_exception, std::type_info *tinfo,
 }
 }
 
+static void handleNSException(NSException *exception) {
+    if (bsg_g_installed) {
+        bool wasHandlingCrash = bsg_g_context->handlingCrash;
+        bsg_kscrashsentry_beginHandlingCrash(bsg_g_context);
+
+        BSG_KSLOG_DEBUG(@"Exception handler is installed. Continuing exception handling.");
+
+        if (wasHandlingCrash) {
+            BSG_KSLOG_INFO(@"Detected crash in the crash reporter. Restoring "
+                           @"original handlers.");
+            bsg_g_context->crashedDuringCrashHandling = true;
+            bsg_kscrashsentry_uninstall((BSG_KSCrashType)BSG_KSCrashTypeAll);
+        }
+
+        BSG_KSLOG_DEBUG(@"Suspending all threads.");
+        bsg_kscrashsentry_suspendThreads();
+
+        BSG_KSLOG_DEBUG(@"Filling out context.");
+        NSArray *addresses = [exception callStackReturnAddresses];
+        NSUInteger numFrames = [addresses count];
+        uintptr_t *callstack = new uintptr_t[numFrames];
+        for (NSUInteger i = 0; i < numFrames; i++) {
+            callstack[i] = [addresses[i] unsignedLongValue];
+        }
+
+        bsg_g_context->crashType = BSG_KSCrashTypeNSException;
+        bsg_g_context->offendingThread = bsg_ksmachthread_self();
+        bsg_g_context->registersAreValid = false;
+        bsg_g_context->NSException.name = strdup([[exception name] UTF8String]);
+        bsg_g_context->crashReason = strdup([[exception reason] UTF8String]);
+        bsg_g_context->stackTrace = callstack;
+        bsg_g_context->stackTraceLength = (int)numFrames;
+
+        BSG_KSLOG_DEBUG(@"Calling main crash handler.");
+        bsg_g_context->onCrash();
+
+        BSG_KSLOG_DEBUG(@"Crash handling complete. Restoring original handlers.");
+        bsg_kscrashsentry_uninstall((BSG_KSCrashType)BSG_KSCrashTypeAll);
```

Comment:
we do not call through to the previous handler here as if we were installed last it is redundant

Reply:
Instead of calling to the last C++ handler, this should use [NSGetUncaughtExceptionHandler](https://developer.apple.com/documentation/foundation/1416853-nsgetuncaughtexceptionhandler?language=objc) and call the NSException handler, as it may be set.

Reply:
I have updated this logic so that we call the `NSUncaughtExceptionHandler` if an `NSException` is triggered, and the original terminate handler if a C++ exception occurs.